### PR TITLE
Api integration ,Design update ,Zustand for onboarding, Common api query converter

### PR DESCRIPTION
**Pull request by Abhinesh-Icanio**
Hi Team, 

In this Pr , i updated the onboarding screen , zustand for the oboardingn datas in local storage, add addQueryParams hooks for dropdown apis & helper function for endpoint handling .

https://zustand.docs.pmnd.rs/middlewares/persist

### CODE REVIEW

File: src/components/AppComponents/AppAutoComplete.jsx [hunk 1]
```diff
@@ -124,15 +135,26 @@ const AppAutoComplete = ({
...
                 return (
                     <>
+
                         <Box
                             {...props}
-                            key={option.id} // Add a key for each option
-                            sx={{ px: 2 }}
+                            key={option[valueParam]}
+                            aria-selected={isSelected ? "true" : "false"}
+                            sx={{
```
Comment (Jothi-icanio) [inline src/components/AppComponents/AppAutoComplete.jsx:147]: can we change this inline style.

File: src/components/AppComponents/AppModalContainer.jsx [hunk 2]
```diff
@@ -11,13 +11,25 @@ import {
 } from "@mui/material";
 
 const StyledPaper = styled(Paper)(({ theme, width, height, align }) => ({
-  width: width || "auto",
+  // width: width || "auto",
```
Comment (Jothi-icanio) [inline src/components/AppComponents/AppModalContainer.jsx:14]: Do we need this commented code

File: src/pages/community/edit-community/index.jsx [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+      .required("Email is required"),
+  }),
+  insuranceDetails: Yup.object().shape({
+    insuranceValue: Yup.string().required("Insurance Value is required"),
+    insuranceCoverage: Yup.string().required("Insurance Coverage is required"),
+  }),
+};
+
+const EditCommunity = ({ onClose }) => {
+  const { data: userData, isLoading } = useGetUserById("98765432345678");
```
Comment (Jothi-icanio) [inline src/pages/community/edit-community/index.jsx:56]: Can we add comment to this

File: src/pages/community/edit-community/index.jsx
Comment (Jothi-icanio) [inline src/pages/community/edit-community/index.jsx:1]: Are we using Grid in this page.

**Review by Jothi-icanio — CHANGES_REQUESTED** [review]

File: src/api/onboardApi.js [hunk 4]
```diff
@@ -10,6 +10,14 @@ const getUsersData = async ({ queryKey }) => {
   });
   return reponse;
 };
+const getUserById = async (id) => {
+  console.log(id, "id");
```
Comment (Jothi-icanio) [inline src/api/onboardApi.js]: can we remove console here

File: src/pages/community/index.jsx [hunk 5]
```diff
@@ -1,334 +1,397 @@
...
+      propertyManager: {
+        name: "",
+        email: "",
+        mobile: "",
+        address: "",
+      },
+    },
+    initialValidationSchema: {
+      communityManager: Yup.object().shape({
+        name: Yup.string().required("Name is required"),
```
Comment (Jothi-icanio) [inline src/pages/community/index.jsx]: Can we move the Validation messages to the utils?

File: src/pages/community/index.jsx [hunk 6]
```diff
@@ -1,334 +1,397 @@
...
-        <AppGrid container spacing={4}>
-            <AppGrid
-                item
-                size={{ xs: 12 }}
-                container
-                justifyContent="space-between"
-                alignItems="center"
+      <AppRowBox>
+        <AppGrid item size={{ xs: 2 }}>
+          {activeStep && !finalStep ? (
```
Comment (Jothi-icanio) [inline src/pages/community/index.jsx]: can we reduce the nested functionality here.

File: src/components/AppComponents/AppAutoComplete.jsx [hunk 7]
```diff
@@ -143,29 +143,38 @@ const AppAutoComplete = ({
...
-
                     },
+
                 },
             }}
             componentsProps={{
                 paper: {
                     sx: {
+                        width: 'auto',
+                        background: '#F7F9FB',
```
Comment (Jothi-icanio) [inline src/components/AppComponents/AppAutoComplete.jsx:189]: can we pick the color from the theme

File: src/pages/community/onboarding/CommunityDetails.jsx
Comment (Jothi-icanio) [inline src/pages/community/onboarding/CommunityDetails.jsx:1]: have to remove the redundant code

File: src/pages/dashboard/TaskTable.jsx [hunk 8]
```diff
@@ -199,23 +201,30 @@ const TaskTable = ({ tableData, loading }) => {
...
-                bgcolor={
-                  row?.status === "COMPLETED" ? "grey.300" : `error.lighter`
-                }
-                borderRadius={"15px"}
-              />
-            );
-          })}
-        </Stack>
+
+        {loading ? (
```
Comment (Jothi-icanio) [inline src/pages/dashboard/TaskTable.jsx:205]: can we use the early return method here?

File: src/pages/community/index.jsx [hunk 9]
```diff
@@ -163,49 +170,60 @@ const CommunityOnboarding = () => {
...
+                        <Button
+                            fullWidth
+                            color="secondary"
+                            onClick={handleBack}
+                            variant="outlined"
+                        >
                             Back
                         </Button>
                     ) : (
                         <div></div>
```
Comment (Jothi-icanio) [inline src/pages/community/index.jsx:210]: return `null` here instead of the `div`

File: src/components/AppComponents/AppAutoComplete.jsx (hunk 7 above)
Comment (Abhinesh-Icanio) [reply src/components/AppComponents/AppAutoComplete.jsx:189]: updated

File: src/components/AppComponents/AppAutoComplete.jsx (hunk 1 above)
Comment (Abhinesh-Icanio) [reply src/components/AppComponents/AppAutoComplete.jsx:147]: updated

File: src/components/AppComponents/AppModalContainer.jsx (hunk 2 above)
Comment (Abhinesh-Icanio) [reply src/components/AppComponents/AppModalContainer.jsx:14]: removed

File: src/pages/community/index.jsx (hunk 9 above)
Comment (Abhinesh-Icanio) [reply src/pages/community/index.jsx:210]: to make the componetn consider there is a fragment we use this wmpty div

**Pull request merged**